inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -298,9 +298,73 @@ bool is_exec_graph(const ngraph::Function& f) {
     return false;
 }
 
+bool resolve_dynamic_shapes(const ngraph::Function& f) {
+    const auto & f_results = f.get_results();
+    if (std::all_of(f_results.begin(), f_results.end(),
+            [](std::shared_ptr<Node> results) { return !results->is_dynamic(); })) {
+        return false;
+    }
+
+    auto f_clone = ngraph::clone_function(f);
+
+    const auto & f_ops = f.get_ordered_ops();
+    const auto & f_clone_ops = f_clone->get_ordered_ops();
+    NGRAPH_CHECK(f_ops.size() == f_clone_ops.size(), "Unexpected get_ordered_ops method behaviour");
+
+    for (size_t id = 0; id < f_ops.size(); ++id) {
+        auto & op = f_ops[id];
+        auto & clone_op = f_clone_ops[id];
+
+        if (auto op_subgraph = std::dynamic_pointer_cast<op::util::SubGraphOp>(op)) {
+            resolve_dynamic_shapes(*op_subgraph->get_function());
+        }
+
+        op->validate_and_infer_types();
+        clone_op->validate_and_infer_types();
+
+        // dynamic_to_static function converts dynamic dimensions to static using
+        // upperbound (get_max_length) dimension value.
+        auto dynamic_to_static = [](const PartialShape & shape) -> PartialShape {
+            if (shape.is_static() || shape.rank().is_dynamic()) {
```

Comment:
How does it work if rank is dynamic? In any case we return the dynamic shape, how do we add such shapes in IR?
Or will we throw an exception?

Reply:
Yes, we will return dynamic shape and finally Serialization transformation will throw an exception with the node name and input/output shapes.
So I don't change logic here.